libtsuba/include/tsuba/RDGTopology.h

```diff
@@ -143,20 +143,26 @@ class KATANA_EXPORT RDGTopology {
   /// Optional field, may not be present depending on the kind of topology this is
   /// Requires backing FileView to be mapped & bound, or the RDGTopology to be filled from memory
   const katana::EntityTypeID* edge_condensed_type_id_map() const {
-    KATANA_LOG_VASSERT(
-        edge_condensed_type_id_map_ != nullptr,
-        "Either this optional field is not present, or the RDGTopology must be "
-        "either bound & mapped, or filled from memory.");
+    if (edge_condensed_type_id_map_size() > 0) {
+      KATANA_LOG_VASSERT(
+          edge_condensed_type_id_map_ != nullptr,
+          "Either this optional field is not present, or the RDGTopology must "
+          "be "
+          "either bound & mapped, or filled from memory.");
+    }
     return edge_condensed_type_id_map_;
   }
 
   /// Optional field, may not be present depending on the kind of topology this is
   /// Requires backing FileView to be mapped & bound, or the RDGTopology to be filled from memory
   const katana::EntityTypeID* node_condensed_type_id_map() const {
-    KATANA_LOG_VASSERT(
-        node_condensed_type_id_map_ != nullptr,
-        "Either this optional field is not present, or the RDGTopology must be "
-        "either bound & mapped, or filled from memory.");
+    if (node_condensed_type_id_map_size() > 0) {
+      KATANA_LOG_VASSERT(
+          node_condensed_type_id_map_ != nullptr,
```

Comment:
This feels weird to me. Why don't we create an empty map so that we avoid `nullptr` dereferences?

I'm fine with adding a TODO for this to get this merged.

Reply:
Yeah, there are definitely better ways to do this; ideally we wouldn't reason about pointers like this at all. Part of the problem is that tsuba is managing its own memory, and doesn't know about things like numa allocation so all of these buffers that tsuba creates are copied into buffers that we think will preform better.

I think an interface where the caller passes tsuba some memory allocator callback would work well actually, then the tsuba code just fill buffers. Not going to land before Thanksgiving though.

Reply:
Makes sense. Thanks.